### ISSUE TITLE
Review Liveness/Readiness probe setup for function's pods

### ISSUE THREAD

**Issue by kwiatekus** [kyma-project/kyma#11203]
**Description**

After introducing readiness and liveness probes for functions ([PR](https://github.com/kyma-project/kyma/pull/10889)) we observe that under the same runtime conditions (memory and CPU resources) the same functions that didn't restart in version `1.20` are now restarting in version `1.21`.

When inspecting the function's pod events we see that kubelet reports unhealthy state based on probes failures. 

~As a workaround, the runtime profile for the functions should be increased ( i.e from `M` to `L` ) which improves it and mitigates the restarting pods.~ EDIT : even scaled functions suffer from non-functioning probes

**Troubleshooting**
`kubectl describe pod`
```
Events:
  Type     Reason     Age                From                       Message
  ----     ------     ----               ----                       -------
  Normal   Scheduled  <unknown>          default-scheduler          Successfully assigned default/xxx to shoot--kyma--xxx
  Normal   Pulled     25m                kubelet, shoot--kyma--xxx  Container image "eu.gcr.io/kyma-project/external/istio/proxyv2:1.8.2-distroless" already present on machine
  Normal   Created    25m                kubelet, shoot--kyma--xxx  Created container istio-init
  Normal   Started    25m                kubelet, shoot--kyma--xxx  Started container istio-init
  Normal   Started    25m                kubelet, shoot--kyma--xxx  Started container istio-proxy
  Normal   Pulled     25m                kubelet, shoot--kyma--xxx  Container image "eu.gcr.io/kyma-project/external/istio/proxyv2:1.8.2-distroless" already present on machine
  Normal   Created    25m                kubelet, shoot--kyma--xxx  Created container istio-proxy
  Normal   Killing    25m                kubelet, shoot--kyma--xxx  Container function failed liveness probe, will be restarted
  Warning  Unhealthy  25m (x3 over 25m)  kubelet, shoot--kyma--xxx  Liveness probe failed: Get http://100.96.3.69:15020/app-health/function/livez: net/http: request canceled (Client.Timeout exceeded while awaiting headers)
  Warning  Unhealthy  25m (x3 over 25m)  kubelet, shoot--kyma--xxx  Readiness probe failed: Get http://100.96.3.69:15020/app-health/function/readyz: net/http: request canceled (Client.Timeout exceeded while awaiting headers)
  Warning  Unhealthy  25m                kubelet, shoot--kyma--xxx  Readiness probe failed: HTTP probe failed with statuscode: 500
  Normal   Pulled     24m (x2 over 25m)  kubelet, shoot--kyma--xxx  Container image "..." already present on machine
  Normal   Started    24m (x2 over 25m)  kubelet, shoot--kyma--xxx  Started container function
  Normal   Created    24m (x2 over 25m)  kubelet, shoot--kyma--xxx  Created container function
  Warning  Unhealthy  24m                kubelet, shoot--kyma--xxx  Liveness probe failed: HTTP probe failed with statuscode: 500
```

**Comment by ptesny** [comment]
Hello,
Looking at code changes that introduced this regression I have a gut feeling this is a race condition scenario.
It looks to me that changing the runtime profile to L can only impact the timing of probe execution. 
And if this is the case that means a race condition ... your thoughts here ?

Since I reported this issue initially to the attention of kyma developers I have experienced random outages (unhealthy stream) with one of my functions even if I assigned to it the XL profile. 

IMO, the best short term option is to revert the code changes that introduced this regression.
best regards
Piotr
